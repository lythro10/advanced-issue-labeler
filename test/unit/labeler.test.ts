import { describe, it, expect, beforeEach, test } from 'vitest';

import {
  ILabelerTestContext,
  labelerContextFixture,
} from './fixtures/labeler.fixture';

describe('IssueForm Object', () => {
  beforeEach<ILabelerTestContext>(context => {
    context.labelers = labelerContextFixture.labelers;
  });

  it<ILabelerTestContext>('can be instantiated', context =>
    context.labelers.map(labelerItem => expect(labelerItem).toBeDefined()));

  test<ILabelerTestContext>('get config()', context =>
    context.labelers.map(labelerItem =>
      expect(labelerItem.config).toMatchSnapshot()
    ));

  test<ILabelerTestContext>('get inputs()', context =>
    context.labelers.map(labelerItem => {
      expect(labelerItem.section).toMatchSnapshot();
      expect(labelerItem.blockList).toMatchSnapshot();
      expect(labelerItem.template).toMatchSnapshot();
    }));

  test<ILabelerTestContext>('get isConfig()', context =>
    context.labelers.map(labelerItem =>
      expect(labelerItem.isConfig).toMatchSnapshot()
    ));

  test<ILabelerTestContext>('get isInputs()', context =>
    context.labelers.map(labelerItem =>
      expect(labelerItem.isInputs).toMatchSnapshot()
    ));

  test<ILabelerTestContext>('gatherLabels()', context =>
    context.labelers.map(labelerItem =>
      expect(labelerItem.gatherLabels()).toMatchSnapshot()
    ));
});
