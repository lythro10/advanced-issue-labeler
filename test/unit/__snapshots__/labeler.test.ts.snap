// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IssueForm Object > gatherLabels() 1`] = `undefined`;

exports[`IssueForm Object > get config() 1`] = `
Config {
  "path": "advanced-issue-labeler.yml",
  "policy": undefined,
}
`;

exports[`IssueForm Object > get inputs() 1`] = `"severity"`;

exports[`IssueForm Object > get inputs() 2`] = `
[
  "",
]
`;

exports[`IssueForm Object > get inputs() 3`] = `""`;

exports[`IssueForm Object > get isConfig() 1`] = `false`;

exports[`IssueForm Object > get isInputs() 1`] = `true`;
